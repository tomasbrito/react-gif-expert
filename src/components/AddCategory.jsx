import { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ( {onNewCategory} ) => {

    const [inputValue, setinputValue] = useState('One Punch')

    const onInputChange = (event) => {
        setinputValue( event.target.value )
    }

    const onSubmit = ( event ) => {
        event.preventDefault()
        console.log(inputValue)
        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim())
        setinputValue('')
    }

    return (
                        //tmb se puede escribir {onSubmit}. Recibe y envia el primer parametro que recibe osea el evento
        <form onSubmit={ (event) => onSubmit(event) } aria-label="form">
            <input type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={(event) => { onInputChange(event) }}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}