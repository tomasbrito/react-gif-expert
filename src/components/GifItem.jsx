import PropTypes from 'prop-types'

export const GifItem = ({title, url, id}) => {
  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
   // {/* <li key={ id }>{ title } </li> */}
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
