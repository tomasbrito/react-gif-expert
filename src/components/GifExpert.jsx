import { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Futbol'])


    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories])
    }


    console.log(categories)


    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory
                onNewCategory={value => onAddCategory(value)}

            />

            {
                categories.map(category => (
                    <GifGrid key= { category } category = { category } />
                ))
            }
        </>
    )
}
