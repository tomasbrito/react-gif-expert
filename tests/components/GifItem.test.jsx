import { GifItem } from "../../src/components/GifItem"
import { render, screen } from "@testing-library/react"



describe('Pruebas en GifItem', () => { 

    const title = 'Fernando'

    test('snapshot', () => { 

        const { container } = render(<GifItem title={ title } url={'www.google.com'}/>)

        expect(container).toMatchSnapshot()
        
     })

     test('should first', () => {  

        render(<GifItem title={ title } url={'www.google.com'}/>)
        expect(screen.getByText(title)).toBeTruthy()
        
        

     })

 })

 