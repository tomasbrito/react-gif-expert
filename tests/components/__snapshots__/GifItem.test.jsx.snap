// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Fernando"
      src="www.google.com"
    />
    <p>
      Fernando
    </p>
  </div>
</div>
`;
